internal/webhooks/test/clusterclass_test.go

```diff
@@ -28,234 +28,250 @@ import (
 	"sigs.k8s.io/cluster-api/internal/test/builder"
 )
 
```

Comment:
Q: Can you give some context on why the old test cases where removed? Are they included in the new ones?

Reply:
Switching to index means that none of the Delete cases works in tests using fakeclient. Indexes only work with testenv.

All of the delete test cases were removed (I moved over the remaining ones in the latest change) and should be covered by tests in the new internal/webhooks/test package.

Reply:
I meant why the cases in this package (which are using envtest) where dropped. I assume they are covered by the new ones.

Reply:
Just took a look - those tests got removed when I was doing the merge / rebase. Thanks for catching that. I've added them back in the latest version of the code.

Reply:
Ah good. I was trying to figure out how they could be part of the new ones :joy: